src/extrapolation/reflect.jl

```diff
@@ -0,0 +1,49 @@
+type ReflectingExtrapolation{T,N,ITP,IT,GT} <: AbstractExtrapolation{T,N,ITP,IT,GT}
+    itp::ITP
+end
+ReflectingExtrapolation{T,ITP,IT,GT}(::Type{T}, N, itp::ITP, ::Type{IT}, ::Type{GT}) =
+    ReflectingExtrapolation{T,N,ITP,IT,GT}(itp)
+
+extrapolate{T,N,IT,GT}(itp::AbstractInterpolation{T,N,IT,GT}, ::Type{Reflect}) =
+    ReflectingExtrapolation(T,N,itp,IT,GT)
+
+function extrap_prep{T,ITP,IT,GT}(etp::Type{ReflectingExtrapolation{T,1,ITP,IT,GT}}, x)
+    quote
+        # translate x_d to inside the domain, and count the translations
+        ntransl = 0
+        while x < $(GT == OnGrid ? 1 : .5)
```

Comment:
If `size(etp.itp, 1) == 10`, this will get really slow for `x = -10^5` :smile:. `Grid` does this using modulus-type operations. You can take the modulus relative to `2*size(etp.itp,1)` and then figure out whether it's even or odd.